parsing/lexer.mll

```diff
@@ -569,8 +609,15 @@ rule token = parse
             { PREFIXOP op }
   | ['~' '?'] symbolchar_or_hash + as op
             { PREFIXOP op }
-  | ['=' '<' '>' '|' '&' '$'] symbolchar * as op
+  | ['=' '<' '|' '&' '$'] symbolchar * as op
             { INFIXOP0 op }
+  | ['>'] symbolcharnodot symbolchar * as op
+    (* ">" is handled above, ">." below *)
+            { INFIXOP0 op }
+  | ">."
+            { if !lex_metaocaml_braces then GREATERDOT
+              else (* only consume symbolchars outside MetaOCaml mode *)
+                INFIXOP0 (">." ^ symbolchars lexbuf) }
```

Comment:
Is it intentional that the `>.` keyword is handled differently than other operator-like keywords? Keywords like `<-` and `->` can be used at the beginning of operators like `<-!` and `->!`, and `fun x ->!x` is a syntax error. But `>.` will always be a keyword, and `>.!` will not be an operator.

Reply:
I'm not the author of this code, but my guess is that we intentionally don't want to use the longest-match rule (as you suggest) to avoid having `>.>.` lexed as an allowed infix operator, which could show up naturally in cases with two nestings of quotation parentheses: `.< .< foo >.>.`. The solution is to change the operator grammar to avoid operators starting with `>.` -- adaptatively.